### ISSUE TITLE
feature request: Add policies/options to deal with unknown rtypes for sub-commands `preview`, `push`

### ISSUE THREAD

**Issue by m-ueberall** [StackExchange/dnscontrol#1258]
(This is _somewhat_ related to/dependent on issue #1257, but requires additional modifications.)
When identifying changes to be made, `dnscontrol` should be able to deal with unknown rtypes instead of simply panicking–either by _ignoring_ or _deleting_ the latter as a result of the selected operation if (supported by the provider in question and) explicitly requested by the user:
```
[2021-09-14T17:46:01+0200] root@desktop01:~/config/dnscontrol# dnscontrol push
******************** Domain: themenzentrisch.de
----- Getting nameservers from: desec
----- DNS Provider: desec...
panic: unparsable record received from deSEC: unknown rtype (OPENPGPKEY) when parsing (mDMEW[…]+HbQEG) domain=(themenzentrisch.de)

goroutine 1 [running]:
github.com/StackExchange/dnscontrol/v3/providers/desec.nativeToRecords({{0xc0007a2e60, 0x44}, {0xc000510b40, 0x1, 0x4}, 0xe10, {0xc000816ce4, 0xa}, {0x0, 0x0}}, ...)
        /home/runner/work/dnscontrol/dnscontrol/providers/desec/convert.go:29 +0x273
github.com/StackExchange/dnscontrol/v3/providers/desec.(*desecProvider).GetZoneRecords(0xc000010018, {0xc000148cd8, 0x13})
        /home/runner/work/dnscontrol/dnscontrol/providers/desec/desecProvider.go:109 +0x148
github.com/StackExchange/dnscontrol/v3/providers/desec.(*desecProvider).GetDomainCorrections(0xc0005ea3c0, 0xc0002ba2d0)
        /home/runner/work/dnscontrol/dnscontrol/providers/desec/desecProvider.go:80 +0x85
github.com/StackExchange/dnscontrol/v3/commands.run({{{{0x14419d2, 0xc}, {0x0, 0x0}, 0x0, {{...}, 0x0}}, {0x0, 0x0}}, {{0x143e86b, ...}}, ...}, ...)
        /home/runner/work/dnscontrol/dnscontrol/commands/previewPush.go:138 +0x8d8
github.com/StackExchange/dnscontrol/v3/commands.Push(...)
        /home/runner/work/dnscontrol/dnscontrol/commands/previewPush.go:92
github.com/StackExchange/dnscontrol/v3/commands.glob..func8.1(0xc0000f6400)
        /home/runner/work/dnscontrol/dnscontrol/commands/previewPush.go:63 +0xa7
github.com/urfave/cli/v2.(*Command).Run(0xc0001a18c0, 0xc000123fc0)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x64a
github.com/urfave/cli/v2.(*App).RunContext(0xc0003044e0, {0x16d3150, 0xc000038100}, {0xc000032040, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x81e
github.com/urfave/cli/v2.(*App).Run(...)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
github.com/StackExchange/dnscontrol/v3/commands.Run({0xc00014c0c0, 0x1440177})
        /home/runner/work/dnscontrol/dnscontrol/commands/commands.go:65 +0x2a9
main.main()
        /home/runner/work/dnscontrol/dnscontrol/main.go:35 +0x134
[2021-09-14T17:48:28+0200] root@desktop01:~/config/dnscontrol# 
```

**Comment by tlimoncelli** [comment]
Since this bug was filed we removed a LOT of the calls to `panic()` in the code.  Can you check if this is still an issue?

**Comment by m-ueberall** [comment]
> Since this bug was filed we removed a LOT of the calls to `panic()` in the code. Can you check if this is still an issue?

Sure! Unfortunately, `dnscontrol` is still not able to at least ignore unknown _rtypes_. Using the current release, I ran the `get-zones`, `preview`, `push` subcommands against a test domain I obtained in the meantime (so no need to anonymize):

```
[2022-08-11T20:55:45+0200] ueberall_l@desktop01:/mirrored/gluster/mounts/shared/development/configuration/dnscontrol% dnscontrol version
dnscontrol v3.18.1 ("70b5b775502327b300e4d138f089c07907ba7cc2") built 10 Aug 22 19:56 CEST
# Obtained required token
[2022-08-11T20:55:57+0200] ueberall_l@desktop01:.../dnscontrol% dnscontrol -v get-zones --format=js desec - schulbeispiel.de >dnsconfig_new.js1
# File 'dnsconfig_new.js1' contains CNAME, TXT, MX, AAAA entries as expected
# !!!! Now added dummy SMIMEA, OPENPGPKEY DNS entries using web-based UI !!!!
[2022-08-11T21:00:57+0200] ueberall_l@desktop01:.../dnscontrol% dnscontrol -v get-zones --format=js desec - schulbeispiel.de >dnsconfig_new.js2
failed GetZone CDP: creds.json entry missing TYPE field
# File 'dnsconfig_new.js2' is empty
# File 'dnsconfig.js' basically contains the same contents as dnsconfig_new.js1 (everything but the two dummy OPENPGPKEY, SMIMEA entries)
[2022-08-11T21:01:07+0200] ueberall_l@desktop01:.../dnscontrol% dnscontrol preview
INFO: In dnsconfig.js NewRegistrar("none", "NONE") can be simplified to NewRegistrar("none") (See https://stackexchange.github.io/dnscontrol/creds-json#cleanup)
WARNING: For future compatibility, update the "desec" entry in creds.json by adding: "TYPE": "DESEC", (See https://stackexchange.github.io/dnscontrol/creds-json#missing)
******************** Domain: schulbeispiel.de
----- Getting nameservers from: desec
----- DNS Provider: desec...
panic: unparsable record received from deSEC: unknown rtype (SMIMEA) when parsing (1 1 1 0000) domain=(schulbeispiel.de)

goroutine 1 [running]:
github.com/StackExchange/dnscontrol/v3/providers/desec.nativeToRecords({{0xc000480c78, 0x4}, {0xc000838440, 0x1, 0x4}, 0xe10, {0xc000480ca0, 0x6}, {0x0, 0x0}}, ...)
        /home/runner/work/dnscontrol/dnscontrol/providers/desec/convert.go:28 +0x28e
github.com/StackExchange/dnscontrol/v3/providers/desec.(*desecProvider).GetZoneRecords(0xc000126008?, {0xc000530fe0, 0x10})
        /home/runner/work/dnscontrol/dnscontrol/providers/desec/desecProvider.go:109 +0x148
github.com/StackExchange/dnscontrol/v3/providers/desec.(*desecProvider).GetDomainCorrections(0xc000418460, 0xc00049c0f0)
        /home/runner/work/dnscontrol/dnscontrol/providers/desec/desecProvider.go:80 +0x85
github.com/StackExchange/dnscontrol/v3/commands.run({{{{0x17cf59d, 0xc}, {0x0, 0x0}, 0x0, {{...}, 0x0}}, {0x0, 0x0}}, {{0x17cc01c, ...}}, ...}, ...)
        /home/runner/work/dnscontrol/dnscontrol/commands/previewPush.go:177 +0xcb8
github.com/StackExchange/dnscontrol/v3/commands.Preview(...)
        /home/runner/work/dnscontrol/dnscontrol/commands/previewPush.go:96
github.com/StackExchange/dnscontrol/v3/commands.glob..func7.1(0xc000438a20?)
        /home/runner/work/dnscontrol/dnscontrol/commands/previewPush.go:28 +0x8b
github.com/urfave/cli/v2.(*Command).Run(0xc000438a20, 0xc000540c00)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.11.1/command.go:169 +0x656
github.com/urfave/cli/v2.(*App).RunContext(0xc000381500, {0x1a0f0e0?, 0xc000130020}, {0xc00012a000, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.11.1/app.go:378 +0xf1e
github.com/urfave/cli/v2.(*App).Run(...)
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.11.1/app.go:251
github.com/StackExchange/dnscontrol/v3/commands.Run({0xc000050600?, 0x17cdb8f?})
        /home/runner/work/dnscontrol/dnscontrol/commands/commands.go:65 +0x2ba
main.main()
        /home/runner/work/dnscontrol/dnscontrol/main.go:35 +0x134
[2022-08-11T21:01:21+0200] ueberall_l@desktop01:.../dnscontrol%
# 'dnscontrol push' basically outputs the same as above (panic), unable to ignore/delete said dummy DNS entries
```

**Comment by tlimoncelli** [comment]
Drat.

Ok, now I see where in the code this happens.  Sadly this isn't a small fix.

**Comment by tlimoncelli** [comment]
Closing.  This is a duplicate of https://github.com/StackExchange/dnscontrol/issues/2432
